daemon/mgr/container_logger.go

```diff
@@ -60,6 +71,44 @@ func (mgr *ContainerManager) SetContainerLogPath(c *Container) {
 	// If the logdriver is json-file, the LogPath should be like
 	// /var/lib/pouch/containers/5804ee42e505a5d9f30128848293fcb72d8cbc7517310bd24895e82a618fa454/json.log
 	if c.HostConfig.LogConfig.LogDriver == "json-file" {
-		c.LogPath = filepath.Join(mgr.Config.HomeDir, "containers", c.ID, "json.log")
+		rootDir, _ := mgr.getLogRootDirFromOpt(c, false)
+		c.LogPath = filepath.Join(rootDir, "json.log")
 	}
 }
+
+func (mgr *ContainerManager) getLogRootDirFromOpt(c *Container, createLogRoot bool) (string, error) {
+	// <pouchd-home-dir>/containers/<cid> as default root dir of container log
+	rootDir := mgr.Store.Path(c.ID)
+
+	cfg := c.HostConfig.LogConfig
+	if cfg == nil {
+		return rootDir, nil
+	}
+
+	logOpt := cfg.LogOpts
+
+	if len(logOpt) == 0 {
+		return rootDir, nil
+	}
+
+	specficRootDir, exist := logOpt[logRootDirKey]
```

Comment:
Maybe we can combine checking `LogConfig` , `LogConfig.LogOpts` and `logRootDirKey` ?

Reply:
fix it.